consensus/polybft/state.go

```diff
@@ -128,7 +129,12 @@ func decodeStateSyncEvent(log *ethgo.Log) (*StateSyncEvent, error) {
 }
 
 func decodeExitEvent(log *ethgo.Log, epoch, block uint64) (*ExitEvent, error) {
-	raw, err := exitEventABI.ParseLog(log)
+	if !exitEventABI.Match(log) {
+		// valid case, not an exit event
+		return nil, errNotAnExitEvent
```

Comment:
Could we do `return nil, false, nil` and use the bool to signal if it's an exit event instead of playing with the error?

Reply:
Hmm, we could, but to me this looks much cleaner. Just my opinion, since you can easily use a predefined error to recognize certain cases. Whatever we agree upon, works for me.

Reply:
If you want my two cents, I agree that if it's a valid event it should not return an err, I suggest `return nil, nil` should suffice, a `nil` `ExitEvent` indicates the event wasn't of that type but no error.

Reply:
Thats totally fine by me. Will change it to return nil in this case.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/852/commits/d9bbc65d412e67074c623708a184a265a2fbbd4c